src/hdskins/java/com/voxelmodpack/hdskins/server/AbstractSkinServer.java

```diff
@@ -0,0 +1,55 @@
+package com.voxelmodpack.hdskins.server;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import com.mojang.authlib.GameProfile;
+import com.mojang.authlib.exceptions.AuthenticationException;
+import com.mojang.authlib.minecraft.MinecraftProfileTexture;
+import com.mojang.authlib.minecraft.MinecraftProfileTexture.Type;
+import com.mojang.authlib.yggdrasil.response.MinecraftTexturesPayload;
+import com.voxelmodpack.hdskins.HDSkinManager;
+import com.voxelmodpack.hdskins.util.CallableFutures;
+import com.voxelmodpack.hdskins.util.IndentedToStringStyle;
+
+import net.minecraft.util.Session;
+
+public abstract class AbstractSkinServer implements SkinServer {
```

Comment:
Why even have an interface if you're just going to give everything an non-modifiable implementation?

Reply:
You have a point there. I'll probably change it back.

Reply:
@Sollce My comment appeared in the wrong place, so I'mma clarify. This was about the `ISkinModifier` comment.

AbstrctSkinServer exists purely to hide the boilerplate around creating an Optional and CompletableFuture in the exact same way for every server type. I didn't want to go changing the outside interface because it seemed fairly ingrained in the rest of the code, though it became less necessary when I looked at hiding the callbacks on CompletableFuture inside SkinServer as well.